### PR TITLE
Add  a default viewer via the metadata key `:portal.viewer/default`

### PR DESCRIPTION
**Pull request by JJ-Atkinson**

### CODE REVIEW

File: src/portal/ui/app.cljs [hunk 1]
```diff
@@ -145,15 +145,21 @@
...
 
+(def viewers-by-name
+  (into {} (map (juxt :name identity) viewers)))
+
 (defn use-viewer [settings value]
   (let [set-settings!      (:set-settings! settings)
         selected-viewer    (:selected-viewer settings)
-        compatible-viewers (filter #((:predicate %) value) viewers)]
+        default-viewer     (get viewers-by-name (:portal.viewer/default (meta value)))
+        viewers            (cons default-viewer viewers)
```
Comment (djblue) [inline src/portal/ui/app.cljs:155]: I think this will keep adding the default viewer to the top of the list causing the dropdown to keep growing.

Comment (djblue) [reply src/portal/ui/app.cljs:155]: Screen Shot 2020-12-03 at 3 35 51 PM

Looks like it's a side effect caused by this react issue.

Comment (JJ-Atkinson) [reply src/portal/ui/app.cljs:155]: Weird. I didn't have that happen during testing. Could be a simple fix of changing `viewers (cons default-viewer viewers)` to `viewers (cons default-viewer (remove #(= default-viewer %) viewers))` if I understand the issue correctly

File: src/examples/default_visualizer.clj [hunk 2]
```diff
@@ -0,0 +1,20 @@
+(ns examples.default-visualizer
+  (:require
+   [clojure.core.protocols :refer [nav]]
+   [clojure.string :as str]))
+
+(defn nav-dep-anno-tree [coll _ v]
+  (let [{:keys [deps-map]} (meta coll)]
+    (with-meta
+      [:div "Depends on "
+       (str/join ", " (map str (get deps-map v)))]
```
Comment (djblue) [inline src/examples/default_visualizer.clj:10]: `(get deps-map k)` is probably what you want.

**Pull request merged**